File: tests/fixtures/javascript/deep_nesting.js
function deeplyNested(data) {
    for (const group of data) {
        if (group.active) {
            for (const item of group.items) {
                if (item.valid) {
                    for (const sub of item.children) {
                        if (sub.ready) {
                            process(sub);
                        }
                    }
                }
            }
        }
    }
    return true;
}

function moderatelyNested(data) {
    for (const item of data) {
        if (item.active) {
            for (const sub of item.children) {
                process(sub);
            }
        }
    }
    return true;
}

function process(x) {}

File: tests/fixtures/r/deep_nesting.r
deeply_nested <- function(data) {
  if (length(data) > 0) {
    for (item in data) {
      if (item > 0) {
        if (item > 100) {
          if (item > 1000) {
            print(item)
          }
        }
      }
    }
  }
}

moderately_nested <- function(x) {
  if (x > 0) {
    x + 1
  } else {
    x - 1
  }
}
